### PR TITLE
Extra configuration for API prefix and token/login endpoints

### PR DESCRIPTION
**Pull request by synbot**
### Acceptance Criteria
1.  `auth-gateway` looks at the config to build the refresh token request
1. If no oauth endpionts are configured the gateway falls back to `/oauth/token` and gives a warning
1. Api requests are prefixed with the `proxy.hostname config` variable if it exists

### CODE REVIEW

**Comment by bobeagan** [comment]
:o:

**Comment by areida** [comment]
:star:

File: UPGRADE_GUIDE.md [hunk 1]
```diff
@@ -1,6 +1,31 @@
...
+        },
+        port   : 9000,
+        prefix : '/api'
+    },
+    login_url : '/login',
+    proxy     : {
+        hostname : 'api.project.vm'
+    }
+};
+```
```
Comment (paulstatezny) [inline UPGRADE_GUIDE.md]: Note to people that see this -- this includes config changes from #98.

**Comment by spruce-bruce** [comment]
:hand:

**Comment by spruce-bruce** [comment]
:+1:

**Comment by paulstatezny** [comment]
:+1:

**Pull request merged**